src/__tests__/components/fields/contact-field/contact-field.spec.tsx

```diff
@@ -152,25 +143,116 @@ describe(UI_TYPE, () => {
+			const contactNumber = "979584362";
 			renderComponent({ validation: [{ contactNumber: { internationalNumber: true } }] });
+
 			await waitFor(() => fireEvent.click(getDefaultDropdownToggle()));
-			const japanCode = getField("button", "Japan (+81)");
-			await waitFor(() => fireEvent.click(japanCode));
 
+			const japanCode = getField("button", "Japan+81");
+
+			await waitFor(() => fireEvent.click(japanCode));
 			fireEvent.change(getContactField(), { target: { value: contactNumber } });
 			await waitFor(() => fireEvent.click(getSubmitButton()));
 
 			expect(SUBMIT_FN).toBeCalledWith(expect.objectContaining({ [COMPONENT_ID]: `+81 ${contactNumber}` }));
 		});
 
-		it("+81 12345678  should be an invalid number", async () => {
-			const contactNumber = "12345678";
+		it("+81 12-345-678 should be an invalid number", async () => {
+			const contactNumber = "12-345-678";
 			renderComponent({ validation: [{ contactNumber: { internationalNumber: true } }] });
+
 			await waitFor(() => fireEvent.click(getDefaultDropdownToggle()));
-			const japanCode = getField("button", "Japan (+81)");
+
+			const japanCode = getField("button", "Japan+81");
+
 			await waitFor(() => fireEvent.click(japanCode));
+			fireEvent.change(getContactField(), { target: { value: contactNumber } });
+			await waitFor(() => fireEvent.click(getSubmitButton()));
+
+			expect(screen.getByText(ERROR_MESSAGES.CONTACT.INVALID_INTERNATIONAL_NUMBER)).toBeInTheDocument();
+		});
+
+		it("should validate against specific country when specified", async () => {
+			const contactNumber = "512345678";
+			const country = "France";
+
+			renderComponent({
+				validation: [
+					{
+						contactNumber: {
+							internationalNumber: country,
+						},
+					},
+				],
+			});
+
+			fireEvent.change(getContactField(), { target: { value: contactNumber } });
+			await waitFor(() => fireEvent.click(getSubmitButton()));
+
+			expect(SUBMIT_FN).toBeCalledWith(expect.objectContaining({ [COMPONENT_ID]: `+33 ${contactNumber}` }));
+		});
+
+		it("should fixed the country selection if singaporeNumber in validation", async () => {
+			renderComponent({
+				validation: [
+					{
+						contactNumber: {
+							singaporeNumber: "mobile",
+						},
+					},
+				],
+			});
+
+			expect(screen.queryByTestId("addon-selector")).not.toBeInTheDocument();
+			expect(screen.getByTestId("addon")).toBeInTheDocument();
+		});
+
+		it("should fixed the specific country when specified in internationalNumber validation", async () => {
+			const country = "Denmark";
+
+			renderComponent({
+				validation: [
+					{
+						contactNumber: {
+							internationalNumber: country,
+						},
+					},
+				],
+			});
+
+			expect(screen.queryByTestId("addon-selector")).not.toBeInTheDocument();
+			expect(screen.getByTestId("addon")).toBeInTheDocument();
+		});
+
+		it("should not fixed the country selection if not specific country internationalNumber validation", async () => {
```

Comment:
`should not fix the country selection if internationalNumber validation does not have a specific country`